### ISSUE TITLE
Type of variable incorrectly narrowed to `Unbound`

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#3387]
Pyright generates a type error in the following code when it should type check fine. It appears to be marking the variable `e` as unbound at the incorrect location in the code flow. It should be unbound only after the `except` block exits.

```python
import os
import errno

def func():
    try:
        pass
    except OSError as e:
        if os.name == "nt" and e.errno == errno.EACCES:
            pass
```

**Comment by erictraut** [comment]
The problem is not that `e` is unbound because of the `except`. The problem is that the subexpression `e.errno == errno.EACCESS` is unreachable because `os.name == "nt"` evaluates to `False`.

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.240, which I just published. It will also be included in the next release of pylance.
